src/BufferingParser.php

```diff
@@ -87,44 +87,49 @@ private function parseBody(string $body, string $boundary = null): Form
             return new Form([]);
         }
 
-        $exp = \explode("\r\n--$boundary\r\n", $body);
+        $exp = \explode("\r\n--$boundary\r\n", $body, $this->fieldCountLimit);
         $exp[0] = \substr($exp[0], \strlen($boundary) + 4);
         $exp[\count($exp) - 1] = \substr(\end($exp), 0, -\strlen($boundary) - 8);
 
         foreach ($exp as $entry) {
-            list($rawHeaders, $text) = \explode("\r\n\r\n", $entry, 2);
-            $headers = [];
-
-            foreach (\explode("\r\n", $rawHeaders) as $header) {
-                $split = \explode(":", $header, 2);
-                if (!isset($split[1])) {
-                    return new Form([]);
-                }
-                $headers[\strtolower($split[0])] = \trim($split[1]);
+            if (($position = \strpos($entry, "\r\n\r\n")) === false) {
+                throw new ParseException("No header/body boundry found");
+            }
+
+            try {
+                $headers = Rfc7230::parseHeaders(\substr($entry, 0, $position + 2));
+            } catch (InvalidHeaderException $e) {
+                throw new ParseException("Invalid headers in body part", 0, $e);
             }
 
+            $entry = \substr($entry, $position + 4);
+
             $count = \preg_match(
                 '#^\s*form-data(?:\s*;\s*(?:name\s*=\s*"([^"]+)"|filename\s*=\s*"([^"]+)"))+\s*$#',
-                $headers["content-disposition"] ?? "",
+                $headers["content-disposition"][0] ?? "",
                 $matches
             );
 
             if (!$count || !isset($matches[1])) {
-                return new Form([]);
+                throw new ParseException("Missing or invalid content disposition");
```

Comment:
This would be the only minor "BC break", right?

Reply:
Yes, it makes `BufferingParser` consistent with `StreamingParser`.

Reply:
I think it's fine.